changelog/3045.added.md

```diff
@@ -0,0 +1,32 @@
+- Introducing user and bot turn start strategies. Turn start strategies are used
+  to indicate when user and bot turns begin. User turn start strategies indicate
+  when the user starts speaking (e.g. using VAD events or when a user says one
+  or more words). Bot turn start strategies indicate when the bot should start
+  speaking (e.g. using an end-of-turn detection model or by observing incoming
+  transcriptions). In conversational agents, these are often referred to as
+  start/stop speaking or turn-taking plans/policies.
+
+  Available interruption strategies:
+    - VADUserTurnStartStrategy (default)
+
+  Avaliable speaking strategies:
+    - TranscriptionBotTurnStartStrategy (default)
+    - TurnAnalyzerBotTurnStartStrategy
```

Comment:
Haven't read the code yet, so pardon me if this comment is just noise, but: would it be possible to detect that the user has configured a turn analyzer on their transport (the old way of specifying a turn analyzer) and if so default to using this strategy?

Reply:
Per our chat: if the user had previously been using a turn analyzer on the transport, turn strategies will be disabled (and a warning to that effect will be logged)

Reply:
Things keep working though, except you might get duplicate `UserStartSpeakingFrame` and `UserStoppedSpeakingFrame`.

Reply:
From convo: since we'd be breaking smart turn for everyone who's following our best practices (using `LLMContext`), the thing to do is send a frame down from the transport saying "hey, automatically switch to the smart turn bot start strategy"). This would be a temporary, for-backwards-compatibility-only mechanism that would be accompanied by very visible deprecation warnings ("please please use the new turn strategies instead of the smart turn analyzer on the transport")

Reply:
Latest thinking:

Print a (very visible) error in the universal user aggregator saying something to the effect of "hey, your transport-specified smart turn analyzer no longer has an effect. use the new turn-analyzer-based bot turn start strategy instead"